honeybee_energy/ventcool/opening.py

```diff
@@ -129,21 +196,34 @@ def from_dict(cls, data):
             "fraction_height_operable": 0.5,  # Fractional number for height operable
             "discharge_coefficient": 0.17,  # Fractional number for discharge coefficient
             "wind_cross_vent": True  # Wind-driven cross ventilation
+            "air_mass_flow_coefficient_closed": 0.001 # Coefficient for mass flow rate
+            "air_mass_flow_exponent_closed": 0.667 # Exponent for mass flow rate
+            "minimum_density_difference_two_way": 1e-3 # Minimum density for two-way flow
             }
         """
         assert data['type'] == 'VentilationOpening', \
             'Expected VentilationOpening dictionary. Got {}.'.format(data['type'])
 
         area_op = data['fraction_area_operable'] if 'fraction_area_operable' in data \
             and data['fraction_area_operable'] is not None else 0.5
-        height_op = data['fraction_height_operable'] if 'fraction_height_operable' in data \
-            and data['fraction_height_operable'] is not None else 1.0
+        height_op = data['fraction_height_operable'] if 'fraction_height_operable' in \
+            data and data['fraction_height_operable'] is not None else 1.0
```

Comment:
Did you check to be sure the default discharge coefficient of 0.17 makes sense for the AFN?  I know that I originally interpreted this assuming the `fraction_height_operable` is the full height of the window (and the discharge coefficient is reduced as a result) but I don't know if the AFN makes the same assumption.

Reply:
@chriswmackey 

Can you clarify how you calculate the default discharge coefficient?

There's no default in EP. Here's what I found from other sources:
- DesignBuilder suggests 0.65 as a "good general purpose setting". 
- CONTAMW (which the EP engineering reference cites) suggests "0.6 for an orifice and slightly higher for other openings in buildings" (so 0.65 seems reasonable for vent_opening). 
- EP sample files has 0.5 for closed opening and 0.6 for completely open window. 

So I could change it to 0.6+ and cite CONTAMW. But let me know how you calculated 0.17 in case there's something else we should account for. 

I should add, since the discharge coefficient is the ratio between actual/ideal air flow, and EP will calculate the ideal flow in the ZoneVentilation and AFN in comparable ways (former with wind/stack kinetic/gravitational potential energy, and latter with pressure) I don't see why they should be radically different.

Reply:
So what I am hearing from you is that the discharge coefficient is NOT supposed to be computed using the full height of the window whenever the AFN is used.  Instead, it assumes that the "height" is the average distance from the center of the opening to the neutral plane (or roughly a quarter of the window's height). I know this because 0.65 is the generally agreed-upon discharge coefficient that should be used when windows are unobstructed (no insect screens) and the height is a quarter of the window height.

I guess this 0.65 is what most people use but it always just seemed really un-intuitive for me to use a quarter of the window height and it was much better to wrap this assumption up with the coefficient.  But so be it. I'll plan to change the honeybee-opensutdio-gem to quarter the height of the window for simple ventilation (we can still keep the `fraction_height_operable` input that we use in the schema) and I'll change every other place where the discharge coefficient is used to be representative of this quartering of the height.

Also, I know that the discharge coefficient for a quarter-height window opening AND insect screens is 0.45.  So what do you think about making the default discharge coefficient here to be 0.45 instead of 0.65?  It seemed better to air on the side of conservative air flow but I know I'm also biased by living in a climate where insect screens are a must-have.

Reply:
I think I'll make this change toady and I am going to make the default 0.45 in order to be conservative with the insect screens.

Reply:
@chriswmackey thanks for explaining this. I agree with the 0.45 value since it's much closer to the established precedent of 0.6 - 0.65, (and I agree we should reduce it to account for the prevalence of insect screens).

Furthermore, having done more research into the discharge coefficient, I found this quote: "For example IEA Annexe 20 Air flow Through Large Openings in Buildings indicates that discharge coefficients reported in the literature vary between 0.3 and 0.8 and that it is not understood what causes the differences" (from DesignBuilder docs), which suggests 0.45 is well within acceptable range, and there's a lot of uncertainty associated with this value anyway. 

Regarding this the window height:
> So what I am hearing from you is that the discharge coefficient is NOT supposed to be computed using the full height of the window whenever the AFN is used. Instead, it assumes that the "height" is the average distance from the center of the opening to the neutral plane (or roughly a quarter of the window's height). I know this because 0.65 is the generally agreed-upon discharge coefficient that should be used when windows are unobstructed (no insect screens) and the height is a quarter of the window height.

I've been trying to find more information on the relationship between the discharge and window/neutral pressure plane heights, and so far I can't find any confirmation of this. 

The EP AFN example files assumes a 0.5 'Window/Door Opening Factor' (the % of the window surface that has an opening), and then assumes an 'Opening/Height/Width Factor' (window fully open) of 1 for the discharge coefficient of 0.6, and an 'Opening/Height/Width Factor' of 0 (window fully closed) for the discharge coefficient of 0.5. So it does seem to be assuming 0.6 for a fully open window opening (which is 50% of the window surface). 

I don't think this assumption effects our model at any rate, since the Simple Window Opening object we are using in the AFN only uses one discharge value, and doesn't require different assumptions for different opening heights. And as I said, the 0.45 is reasonable according to multiple reference I've listed above.

Reply:
FYI, I found this reference in the DB docs that helped me understand that the discharge coefficient is actually closely related to the flow coefficient we're using in the Crack model. Specifically, it's based on a different variation of flow rates derived from Bernoulli's equation: 

![image](https://user-images.githubusercontent.com/1685956/90103100-f69db500-dd0f-11ea-99e0-9755711dfefd.png)

So you can see there that in the orifice equation used for window openings, the 'flow exponent' is square root (0.5), and the Cd*A is serving the same role as the flow-coefficient.

Reply:
The comment I made about the quartering of the window height had more to do with how we should be translating the data about the discharge coefficient for the [simple WindAndStack ventilation objects](https://bigladdersoftware.com/epx/docs/9-3/input-output-reference/group-airflow.html#zoneventilationwindandstackopenarea).  We just want the user input to be consistent between the AFN and the simpler ZoneVentilation where possible.  You can see at the link there that the E+ Input/Output reference basically interprets the discharge coefficient in a consistent manner between AFN and zone ventilation, suggesting that I use the "height from midpoint of lower opening to the neutral pressure level" for the simple objects (the quarter height).  So I made the change in the honeybee-openstudio-gem as we translate the schema input over to OSM format:
https://github.com/ladybug-tools/honeybee-openstudio-gem/pull/134

Because the AFN is much more complex in the the way it models stack effects, we shouldn't have to do this process of quartering the height.  But all of this means that we can keep a consistent definition of discharge coefficients around the 0.3-0.8 range.